pkg/ipmanager/ipam/controller.go

```diff
@@ -0,0 +1,236 @@
+package ipam
+
+import (
+	"fmt"
+	"sync"
+	"time"
+
+	k8sv1 "k8s.io/api/core/v1"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilruntime "k8s.io/apimachinery/pkg/util/runtime"
+	"k8s.io/apimachinery/pkg/util/wait"
+	"k8s.io/client-go/tools/cache"
+	"k8s.io/client-go/util/retry"
+	"k8s.io/client-go/util/workqueue"
+	"k8s.io/klog"
+)
+
+func NewController(spec SubmarinerIpamControllerSpecification, config InformerConfigStruct) *Controller {
```

Comment:
The params could be pointers to avoid copy.

Reply:
Done